st2common/st2common/metrics/drivers/prometheus.py

```diff
@@ -0,0 +1,48 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from prometheus_client import Histogram, Gauge
+
+from st2common.metrics.metrics import BaseMetricsDriver
+
+
+class PrometheusDriver(BaseMetricsDriver):
+    """ Base class for driver implementations for metric collection
+    """
+    def __init__(self):
+        pass
+
+    def time(self, key, time):
+        """ Timer metric
+        """
+        prometheus_histogram = Histogram(  # pylint: disable=no-value-for-parameter
```

Comment:
Just curious why that is needed - in theory, if prometheus_client is in PYTHONPATH, pylint should detect the properties (unless the code is doing some magic).

Reply:
Their library is magic and people complain about it and they don't care. :'(